apps/web/src/components/layout/components/v2/HeaderNav.tsx

```diff
@@ -0,0 +1,210 @@
+import { useSpotlightContext } from '../../../providers/SpotlightProvider';
+import { HEADER_HEIGHT } from '../../constants';
+import { NotificationCenterWidget } from '../NotificationCenterWidget';
+
+const FALLBACK_AVATAR = CONTEXT_PATH + '/static/images/avatar.png';
+
+const getThemeTitle = (themeStatus: string) => {
+  if (themeStatus === 'dark') {
+    return 'Dark Theme';
+  } else if (themeStatus === 'light') {
+    return 'Light Theme';
+  } else {
+    return 'Match System Appearance';
+  }
+};
+
+const menuItem = (iconColor: string) => [
+  {
+    title: 'Settings',
+    icon: <IconSettings color={iconColor} />,
+    path: ROUTES.SETTINGS,
+  },
+  {
+    title: 'Invite Members',
+    icon: <IconOutlineGroupAdd color={iconColor} />,
+    path: ROUTES.TEAM,
+  },
+];
+
+const Icon = () => {
+  const { themeStatus } = useLocalThemePreference();
+
+  if (themeStatus === 'dark') {
+    return <IconDarkMode />;
+  }
+  if (themeStatus === 'light') {
+    return <IconLightMode />;
+  }
+
+  return <IconTonality />;
+};
+
+export function HeaderNav() {
+  const { currentOrganization, currentUser, logout } = useAuthContext();
+  const { colorScheme, toggleColorScheme } = useMantineColorScheme();
+
+  const { addItem, removeItems } = useSpotlightContext();
+  const isSelfHosted = IS_DOCKER_HOSTED;
+  const isDark = colorScheme === 'dark';
+  const iconColor = isDark ? colors.white : colors.B40;
+
+  useBootIntercom();
+  useTrackThemeChange();
+
+  const themeTitle = useMemo(() => getThemeTitle(colorScheme), [colorScheme]);
+
+  useEffect(() => {
+    const spotlightMenuItems = [
+      {
+        id: 'toggle-theme',
+        title: themeTitle,
+        icon: <Icon />,
+        onTrigger: () => {
+          toggleColorScheme();
+        },
+      },
+      {
+        id: 'sign-out',
+        title: 'Sign out',
+        icon: <IconLogout />,
+        onTrigger: () => {
+          logout();
+        },
+      },
+    ];
+    removeItems(spotlightMenuItems.map((item) => item.id));
+
+    addItem(spotlightMenuItems);
+  }, [addItem, removeItems, themeTitle, toggleColorScheme, logout]);
+
+  const menuItems = useMemo(() => menuItem(iconColor), [iconColor]);
+
+  const profileMenuMantine = [
+    <Dropdown.Item disabled key="user">
+      <Group spacing={16} noWrap>
+        <Avatar radius="sm" size={40} src={currentUser?.profilePicture || CONTEXT_PATH + '/static/images/avatar.png'} />
+        <div style={{ flex: 1 }}>
+          <Text data-test-id="header-dropdown-username" rows={1} weight="bold" transform="capitalize">
+            {currentUser?.firstName as string} {currentUser?.lastName as string}
+          </Text>
+          <Text
+            size={14}
+            color={colors.B60}
+            rows={1}
+            data-test-id="header-dropdown-organization-name"
+            transform="capitalize"
+          >
+            {currentOrganization?.name as string}
+          </Text>
+        </div>
+      </Group>
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** This could be reusable other places, and adds a lot of JSX to an-already big component. Perhaps this can be its own component outside of this big chunk?‏